frontend/src/features/homes/add-home.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+  <q-dialog
+    class="fit"
+    v-model="show"
+    @hide="$emit('close')"
+    @escape-key="$emit('close')"
+  >
+    <q-card>
+      <q-card-section>
+        <div class="text-h4">
+          <q-icon name="fa fa-cubes" class="q-mr-md" />
+          <span v-if="home">
+            {{ $i18n.t("editHome-title") }}
+          </span>
+          <span v-else>
+            {{ $i18n.t("newHome-header-text") }}
+          </span>
+        </div>
+      </q-card-section>
+
+      <q-separator />
+      <q-card-section>
+        <q-form @submit="onSubmit" ref="manageHome" class="q-gutter-md">
+          <q-input
+            v-model="homeName"
+            :rules="[v => !!v || $i18n.t('fieldRequired')]"
```

Comment:
Put this in a `validateRequiredField` function that can be used in other places. At the very least it makes the code more readable, while also allowing re-use (even though the function is fairly simple).

Reply:
moved to validator utils